src/features/auth/components/LoginSettings.tsx

```diff
@@ -1,53 +1,64 @@
 import { Button } from "components/ui/Button";
 import { Modal } from "components/ui/Modal";
 import { CloseButtonPanel } from "features/game/components/CloseablePanel";
-import { AppearanceSettings } from "features/island/hud/components/settings-menu/general-settings/AppearanceSettings";
-import { LanguageSwitcher } from "features/island/hud/components/settings-menu/general-settings/LanguageChangeModal";
 import { useAppTranslation } from "lib/i18n/useAppTranslations";
 import React, { useState } from "react";
 
 import settingsIcon from "assets/icons/settings_disc.png";
 import { PIXEL_SCALE } from "features/game/lib/constants";
 import { createPortal } from "react-dom";
+import {
+  SettingMenuId,
+  settingMenus,
+} from "features/island/hud/components/settings-menu/GameOptions";
 
 export const LoginSettings: React.FC = () => {
   const [showModal, setShowModal] = useState(false);
-  const [showLanguage, setShowLanguage] = useState(false);
-  const [showFonts, setShowFonts] = useState(false);
+  const [currentMenu, setMenu] = useState<SettingMenuId | "login">("login");
+
+  const buttons: SettingMenuId[] = ["changeLanguage", "appearance"];
 
   const { t } = useAppTranslation();
 
   const close = () => {
-    setShowLanguage(false);
-    setShowFonts(false);
     setShowModal(false);
+    setMenu("login");
   };
-  const Content = () => {
-    if (showFonts) {
-      return <AppearanceSettings />;
-    }
 
-    if (showLanguage) {
-      return <LanguageSwitcher />;
+  const Content = () => {
```

Comment:
Hey @eliasSFL, I know you didn't create this component but now is a good time to refactor this as creating a "component" inside another component is generally bad practice. You can read about it [here](https://dev.to/sanjampreetsingh/nested-components-a-bad-practice-reactjs-1hbf). 

In this case this code inside of `Content` can just be moved inline in the JSX return like the below example:

```
{currentMenu !== "login" && <code>}
{currentMenu === "login" && <code>}
```